apps/extension/src/ui/domains/Mnemonic/Verify.tsx

```diff
@@ -0,0 +1,157 @@
+  const activeIndex = matchedWords.length
+  const [errorIndex, setErrorIndex] = useState<number>()
+
+  const decoyWords = useMemo(() => {
+    if (!mnemonicWords) return []
+    const decoys = []
+    while (decoys.length < 12) {
+      const index = Math.floor(Math.random() * wordlist.length)
+      if (!mnemonicWords.includes(wordlist[index])) decoys.push(wordlist[index])
+    }
+    return decoys
+  }, [mnemonicWords])
+
+  const displayWords = useMemo(() => {
+    if (!mnemonicWords) return []
+    return [...mnemonicWords, ...decoyWords].sort()
+  }, [mnemonicWords, decoyWords])
+
+  const handleSelectWord = useCallback(
+    (word: string) => {
+      if (!mnemonicWords) return
+      setErrorIndex(undefined)
+      setSelectedWord(word)
+      if (mnemonicWords[activeIndex] === word) {
+        setMatchedWords((prev) => [...prev, word])
+      } else {
+        setErrorIndex(activeIndex)
+      }
+    },
+    [activeIndex, mnemonicWords]
+  )
+
+  if (!mnemonic) return <>No Mnemonic Available</>
+
+  return (
+    <div>
+      <div className="flex flex-col gap-12">
+        <div className="flex flex-col gap-8">
+          <span className="text-body-secondary text-xs">
+            {t("Confirm your recovery phrase by selecting the words below.")}
+          </span>
+          <div className="bg-black-secondary group relative overflow-hidden rounded p-2">
+            <div
+              className={`grid min-h-[12.6rem] grid-cols-4 ${
+                mnemonicWords!.length > 12 && "lg:grid-cols-6"
+              } gap-4 p-2`}
+            >
+              {!!mnemonicWords &&
+                mnemonicWords.map((_, i) => (
+                  <WordSlot
+                    number={i + 1}
+                    word={errorIndex === i ? selectedWord : matchedWords[i]}
+                    active={activeIndex === i}
+                    error={errorIndex === i}
+                    key={`mnemonic-${i}`}
+                  />
+                ))}
+            </div>
+          </div>
+          <div className="flex flex-wrap justify-center gap-4 p-2">
+            {!!displayWords &&
+              displayWords.map((word, i) => (
+                <WordOption
+                  key={`decoyWords-${i}`}
+                  onClick={handleSelectWord}
+                  word={word}
+                  selected={matchedWords.includes(word)}
+                />
+              ))}
+          </div>
+        </div>
+        <div className="flex flex-col gap-4">
+          <Button
+            primary
+            onClick={handleComplete}
+            disabled={mnemonicWords && matchedWords.length < mnemonicWords.length}
+            tabIndex={0}
+          >
+            {t("Complete Verification")}
+          </Button>
+          <button
+            className="text-body-secondary hover:text-grey-300 flex cursor-pointer items-center gap-4 text-sm"
+            onClick={handleBack}
+            onKeyDown={handleBack}
```

Comment:
might want to remove onKeyDown, i assume it was put there to prevent warnings while it was a div